### PR TITLE
Work around netlink bug on ipset create error

### PR DESCRIPTION
**Pull request by bleggett**
**Please provide a description of this PR:**

1. Normally, we unconditionally remove the `ipset` we create in istio cni on shutdown
2. if istio cni happens to crash or otherwise terminate uncleanly, there is a chance we can not remove the ipset
3. this is fine, as on startup we explicitly ignore `IPSET_ERR_EXIST` errors from `netlink` so we can recover.

Except it is not fine, as the error `netlink` ACTUALLY returns is not that at all, which _appears_ to be due to an early return: https://github.com/vishvananda/netlink/blob/d13535d71ed3e430e26c7538cd6266d026b26f54/ipset_linux.go#L410

and we consequently do not ignore.

So, do something a little more foolproof

`netlink` bug is: https://github.com/vishvananda/netlink/issues/992

We have  been doing this check since 1.21 at least, and `netlink` does the same kind of check in their test code - it just doesn't actually work correctly in all cases.

Will see if I can add an integ test for this as well.

You can repro this by 

- manually jumping onto a node before installing istio
- manually creating a `hash:ip` ipset named `istio-inpod-probes-v4`
- installing istio

### CODE REVIEW

File: cni/pkg/ipset/nldeps_linux.go [hunk 1]
```diff
@@ -40,10 +42,19 @@ func (m *realDeps) ipsetIPHashCreate(name string, v6 bool) error {
 		family = unix.AF_INET
 	}
 	err := netlink.IpsetCreate(name, "hash:ip", netlink.IpsetCreateOptions{Comments: true, Replace: true, Family: family})
-	if ipsetErr, ok := err.(nl.IPSetError); ok && ipsetErr == nl.IPSET_ERR_EXIST {
+	log.Debugf("IPSET ERROR IS: %s", err)
```
Comment (howardjohn) [inline cni/pkg/ipset/nldeps_linux.go:45]: nit: probably no need for caps

**Review by howardjohn — APPROVED** [review]

File: cni/pkg/ipset/nldeps_linux.go (hunk 1 above)
Comment (bleggett) [reply cni/pkg/ipset/nldeps_linux.go:45]: yeah this is cruft ty.

**Comment by bleggett** [comment]
Took a little bit longer than I expected to get the test to actually do what I wanted, but should be good.

Also CNI_NET_DIR is removed from the configmap - it is no longer actually used for anything - added relnotes as this is user-visible.

**Pull request merged**